### ISSUE TITLE
Create issue and PR templates

### ISSUE THREAD

**Issue by rhys-vdw** [tgriesser/knex#1415]
https://github.com/blog/2111-issue-and-pull-request-templates

@jurko-gospodnetic's list of questions [here](https://github.com/tgriesser/knex/issues/1407#issuecomment-219266962) are a good starting point for the issue template.

PR's typically need:

- Tests (where possible/appropriate)
- Reference to any issues that would be closed by the PR.
- Documentation updates (where appropriate)

@elhigu I believe we discussed this previously. Any other points you'd like to add?

**Comment by elhigu** [comment]
@rhys-vdw Sounds good. Here are couple more points:

You mentioned possibility of having changelog always up-to-date for making releases easier. So for pull requests changelog updates would be also nice to have.

Also when implementing a new feature, would be good to discuss about API / implementation beforehands to get some agreement on details how stuff should work.

**Comment by rhys-vdw** [comment]
> So for pull requests changelog updates would be also nice to have.

Added.

> Also when implementing a new feature, would be good to discuss about API / implementation beforehands to get some agreement on details how stuff should work.

I agree, but by the time they're opening the PR it's probably too late for this... Perhaps a warning that API changes may need revision if they haven't been previously discussed.

**Comment by elhigu** [comment]
Yep, better to keep process in bare minimum... IIRC discussing about new stuff has been already mentioned in contributing.md.

**Comment by wubzz** [comment]
+1 for changelogs being added consecutively in each respective pull request. Prevents that awkward one-out commit when publishing and that way the person who did the change -- and ultimate the person who most likely understood the change the most -- gets to describe their change.

**Comment by mercmobily** [comment]
This issue has been partially resolved. An issue template was successfully created and merged in November 2017 (commit 4b9e44f8), which is currently located at `.github/ISSUE_TEMPLATE.md`. However, the pull request template that was discussed in this issue has not yet been implemented.

Looking at the current issue template at `.github/ISSUE_TEMPLATE.md:1-42`, it provides a comprehensive structure for bug reports, feature requests, documentation issues, and redirects questions to appropriate channels. The template includes important elements such as environment details (Knex version, database version, OS), specific sections for different issue types, and guidance on who to tag for dialect-specific issues.

To fully resolve this issue, a pull request template should be created at `.github/PULL_REQUEST_TEMPLATE.md` incorporating the requirements discussed here. This would complement the existing issue template and provide consistent guidance for all contributors.

The fact that this issue has remained open since May 2016 even though the issue template was implemented suggests the PR template component was overlooked or deprioritised. Given the maintainers' expressed interest in having changelog updates and proper documentation with PRs, completing this work would benefit the project's contribution process.

Related Issues:
- #1407: Referenced for the initial list of questions that informed the issue template structure
